js/client-chat.js

```diff
@@ -93,7 +93,9 @@
 				this.tabComplete.reset();
 				this.chatHistory.push(text);
 				text = this.parseCommand(text);
-				if (text) {
+				if (this.id.slice(0, 7) === 'battle-' && this.battle.ignoreSpects && app.user.get('userid') !== this.battle.p1.id && app.user.get('userid') !== this.battle.p2.id) {
```

Comment:
I was remembering it causing issues when it actually was `room.battle` that was the problem and ended up not making that change, so yeah...